prov/rxm/src/rxm_init.c

```diff
@@ -177,6 +177,14 @@ int rxm_info_to_rxm(uint32_t version, const struct fi_info *core_info,
 	return 0;
 }
 
+static inline int rxm_auto_progress_requested(void)
+{
+	char *param;
+
+	return (fi_param_get_str(&rxm_prov, "default_data_progress", &param) ?
+		0 : !strcmp(param, "auto"));
+}
```

Comment:
This isn't needed.  The app already has a way to specify what it requires through fi_getinfo.  If the app doesn't, then the provider should simply report what is optimal for it.  We shouldn't need another control for what's already controllable from the API directly.

Reply:
I agree but this is mostly for testing purpose. Since RxM returns only manual progress mode if the app doesn't explicitly specify one in hints, it's hard to test auto-progress for apps like MPI which leave the progress mode undefined in hints. Some MPI may have run-time options that set the progress mode but having an env variable in rxm makes it a one-stop solution.

Reply:
We don't want to add environment variables unless they are absolutely needed.  It's just more code that needs to be maintained and tests.  (Yes, we should validate the env vars do what they are supposed to).  Please drop this patch.